pages/learn/develop/container-contracts.md

```diff
@@ -0,0 +1,90 @@
+
+```yaml
+type: 'sw.container'
+slug: 'enforce-os-gt-240'
+name: 'Enforce hostOS Version'
+requires:
+  - type: 'sw.os'
+    version: '>=2.40.0'
+```
+
+* `type` should be set to `sw.container` for container contracts.
+* `slug` should be a unique identifier to identify the contract.
+* `name` should be a descriptive name of the contract.
+* `requires` should be a list of requirements to be enforced by the contract defined by `type` and `version` keys. Currently, `type` may be either `sw.os` for the hostOS version or `sw.supervisor` for the device Supervisor version. 
+
+__Note__ If `name` or `type` are not supplied, the build will fail. If `slug` is not provided, the build will complete, but the contract will fail validation on deployment. If `requires` is not provided, the contract will have no effect and the release will be deployed.
+
+Multiple contract requirements may be added to a single contract. The following example requires that the hostOS is greater than or equal to 2.44.0, and the Supervisor version is greater than 10.
+
+```yaml
+type: 'sw.container'
+slug: 'enforce-supervisor-and-os'
+name: 'Enforce hostOS and supervisor'
+requires:
+  - type: 'sw.supervisor'
+    version: '>10'
+  - type: 'sw.os'
+    version: '>=2.44.0'  
+```
+
+__Note__ Types in the contract requirements are not validated, so passing values for `type` other than `sw.os` or `sw.supervisor` will result in the contract passing.
+
+When deploying a release, if the contract requirements are not met, the release will fail, and all devices in the application will remain on their current release. This default behavior can be overridden by including an [optional label](#optional-contracts) in the application docker-compose.yml file. If there are any existing running services, they will not be destroyed and continue running the previous release.
+
+Should a contract fail, the logs will contain detail about the failing contract:
+
+```shell
+Could not move to new release: Some releases were rejected due to having unmet requirements:
+Contracts: Services with unmet requirements: first-service
+```
+
+__Note__ For application fleets that contain a mixture of hostOS and Supervisor versions, contracts may be used to ensure only compatible devices run specific services as releases will only be deployed to those meeting the contract requirements.
+
+## Optional Contracts
+
+By default, when a service contract fails, none of the services are deployed to the device. However, in a multi-container application, it is possible to ignore those services that fail the contract requirements with the other services being deployed as normal. To do so, we make use of [Supervisor labels](/learn/develop/multicontainer/#labels) to indicate which services should be considered optional.
```

Comment:
We need to add the label to this section too

Reply:
Something like the below, leaving the larger explanation to the dedicated section?

```suggestion
By default, when a service contract fails, none of the services are deployed to the device. However, in a multi-container application, it is possible to ignore those services that fail the contract requirements with the other services being deployed as normal. To do so, we make use of the `io.balena.features.optional=1` [Supervisor label](/learn/develop/multicontainer/#labels) to indicate which services should be considered optional.
```